### PR TITLE
recent activity update for apptools and publishing/unpublishing

### PR DESCRIPTION
**Pull request by denis-yuen**
**Description**
Adds UI support for github app tools (pretty ugly though) 
Also publishing and unpublishing events because I got confused as to why they weren't showing up

Blocks on https://github.com/dockstore/dockstore/pull/4778

Screenshot from 2022-03-09 15-17-51

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-4041
https://ucsc-cgl.atlassian.net/browse/SEAB-3787

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain

### CODE REVIEW

**Comment by denis-yuen** [comment]
Would probably look better with templates like https://github.com/dockstore/dockstore-ui2/pull/1481 
But should probably move on, so will create follow-up issue

**Comment by denis-yuen** [comment]
SonarCloud isn't nearly as thorough with UI code as opposed to back-end

File: src/app/home-page/recent-events/recent-events.component.html [hunk 1]
```diff
@@ -31,10 +40,19 @@
             <mat-divider *ngIf="!last" class="custom-margin"></mat-divider>
           </div>
           <div *ngSwitchCase="EventType.ADDTOCOLLECTION">
-            <strong>{{ event.initiatorUser.username }} </strong> added the {{ event?.tool ? 'tool' : 'workflow' }}
-            <a [routerLink]="event?.tool ? '/tools/' + event?.tool.tool_path : '/workflows/' + event?.workflow.full_workflow_path">{{
-              (event?.tool | entryToDisplayName) || (event?.workflow | entryToDisplayName)
-            }}</a>
+            <strong>{{ event.initiatorUser.username }} </strong> added the {{ event?.tool || event?.apptool ? 'tool' : 'workflow' }}
+            <a
```
Comment (coverbeck) [inline src/app/home-page/recent-events/recent-events.component.html:44]: Could avoid duplication with a component or pipe. But that can wait for 1.13.

**Review by coverbeck — APPROVED** [review]

**Review by NatalieEO — APPROVED** [review]

**Review by svonworl — APPROVED** [review]

File: src/app/home-page/recent-events/recent-events.component.html (hunk 1 above)
Comment (denis-yuen) [reply src/app/home-page/recent-events/recent-events.component.html:44]: https://ucsc-cgl.atlassian.net/browse/SEAB-4056

**Pull request merged**